src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -0,0 +1,188 @@
+'''
+Contains helpers that are used to get statistics (specifically number of
+regular files and directories and total size of data present under a given
+directory) and pass, print, log and convert them to human readable format
+conveniently.
+'''
+from logging import getLogger
+from uuid import uuid4
+
+from .operations.volume import open_volume
+
+from mgr_util import RTimer
+
+log = getLogger(__name__)
+
+
+def get_size_h(num):
+    '''
+    Convert size to human-readable format.
+    '''
+    size = ('B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB')
+
+    if isinstance(num, bytes):
+        num = int(num)
+
+    if isinstance(num, str):
+        if not num.isdigit():
+            return num
+        num = int(num)
+
+    if num < 1024:
+        return f'{num} {size[0]}'
+
+    i = 0
+    while True:
+        num = num / 1024
+        i += 1
+        if i >= len(size):
+            log.error(f"Biggest suffix we have is '{len(size)-1}' "
+                       'but this is not sufficient in this case. '
+                      f'Current size = {num}')
+
+        x, y = str(num).split('.')
+        if len(x) <= 3:
+            if y:
+                y = '' if y == '0' else y[:4]
+
+            return f'{x}.{y} {size[i]}' if y else f'{x} {size[i]}'
+
+
+def get_num_h(num):
+    '''
+    Convert number to human-readable format.
+    '''
+    size = ('', 'K', 'M', 'B', 'T')
+
+    if isinstance(num, bytes):
+        num = int(num)
+
+    if isinstance(num, str):
+        if not num.isdigit():
+            return num
+        num = int(num)
+
+    if num < 1000:
+        return f'{num}'
+
+    i = 0
+    while True:
+        num = num / 1000
+        i += 1
+        if i >= len(size):
+            log.error(f"Biggest suffix we have is '{len(size)-1}' "
+                       'but this is not sufficient in this case. '
+                      f'Current size = {num}')
+
+        x, y = str(num).split('.')
+        if len(x) <= 3:
+            if y:
+                y = '' if y == '0' else y[:4]
+
+            return f'{x}.{y} {size[i]}' if y else f'{x} {size[i]}'
+
+
+def get_size_ratio_str(size1, size2, human=True):
+    if human:
+        size1, size2 = get_size_h(size1), get_size_h(size2)
+
+    return f'{size1}/{size2}'
+
+
+def get_num_ratio_str(num1, num2, human=True):
+    if human:
+        num1, num2 = get_num_h(num1), get_num_h(num2)
+
+    return f'{num1}/{num2}'
+
+
+def get_amount_copied(src_path, dst_path, fsh, human=True):
+    rbytes = 'ceph.dir.rbytes'
+
+    size_t = int(fsh.getxattr(src_path, rbytes))
+    size_c = int(fsh.getxattr(dst_path, rbytes))
+
+    if size_t == 0:
+        return -1 if size_c == 0 else -2
+
+    percent = ((size_c/size_t) * 100)
+    percent = round(percent, 3)
+
+    return size_t, size_c, percent
+
+
+def get_stats(src_path, dst_path, fsh, human=True):
+    rfiles = 'ceph.dir.rfiles'
```

Comment:
I don't remember agreeing. Perhaps I forgot some part of that discussion. If we 15 files where 5 are regfiles, 5 dirs and 5 links. Telling user only about 5 regfiles is better than telling user about all 15 files because 5 regfiles will take much more time to be cloned compared to 10 non-regfiles.

However, I've made the change you've requested.

Reply:
> I don't remember agreeing. Perhaps I forgot some part of that discussion. If we 15 files where 5 are regfiles, 5 dirs and 5 links. Telling user only about 5 regfiles is better than telling user about all 15 files because 5 regfiles will take much more time to be cloned compared to 10 non-regfiles.

No, it's not better. You are assuming that there would be a close to equal distribution of regular files, directories and symbolic links.

Reply:
>  You are assuming that there would be a close to equal distribution 

That's just by accident. Even if we have 6 files, 17 directories and 9 symlinks, IIUC, my arguments holds true.

Reply:
BTW - https://github.com/ceph/ceph/pull/54620#issuecomment-1909483109